src/types/form_logic.ts

```diff
@@ -51,3 +52,55 @@ export interface IPreventSubmitLogic extends ILogic {
 export interface IPreventSubmitLogicSchema
   extends IPreventSubmitLogic,
     Document {}
+
+type LogicField = Extract<
+  BasicField,
+  | BasicField.Dropdown
+  | BasicField.Radio
+  | BasicField.YesNo
+  | BasicField.Number
+  | BasicField.Decimal
+  | BasicField.Rating
+>
+
+type LogicCondition<K extends LogicField, V extends LogicConditionState> = [
+  K,
+  Array<V>,
+]
+
+// Logic fields that are categorical
+type CategoricalLogicField = Extract<
+  BasicField,
+  BasicField.Dropdown | BasicField.Radio
+>
```

Comment:
would this be simpler? `Extract` doesn't seem to be helping much
```suggestion
type CategoricalLogicField = BasicField.Dropdown | BasicField.Radio
```